<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1130</width>
    <height>896</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QTableWidget" name="tableWidget">
    <property name="geometry">
     <rect>
      <x>950</x>
      <y>0</y>
      <width>181</width>
      <height>811</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>11</pointsize>
     </font>
    </property>
    <property name="sizeAdjustPolicy">
     <enum>QAbstractScrollArea::AdjustIgnored</enum>
    </property>
    <property name="autoScroll">
     <bool>true</bool>
    </property>
    <property name="showGrid">
     <bool>true</bool>
    </property>
    <attribute name="horizontalHeaderVisible">
     <bool>true</bool>
    </attribute>
    <attribute name="horizontalHeaderCascadingSectionResizes">
     <bool>false</bool>
    </attribute>
    <attribute name="horizontalHeaderDefaultSectionSize">
     <number>50</number>
    </attribute>
    <attribute name="horizontalHeaderMinimumSectionSize">
     <number>47</number>
    </attribute>
    <attribute name="verticalHeaderDefaultSectionSize">
     <number>21</number>
    </attribute>
    <row>
     <property name="text">
      <string>1</string>
     </property>
     <property name="font">
      <font>
       <pointsize>8</pointsize>
      </font>
     </property>
    </row>
    <row>
     <property name="text">
      <string>2</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>3</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>4</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>5</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>6</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>7</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>8</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>9</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>10</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>11</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>12</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>13</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>14</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>15</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>16</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>17</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>18</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>19</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>20</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>21</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>22</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>23</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>24</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>25</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>26</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>27</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>28</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>29</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>30</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>31</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>32</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>33</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>34</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>35</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>36</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>37</string>
     </property>
    </row>
    <column>
     <property name="text">
      <string>Piston</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>Tip</string>
     </property>
     <property name="font">
      <font>
       <pointsize>11</pointsize>
      </font>
     </property>
    </column>
    <column>
     <property name="text">
      <string>Tilt</string>
     </property>
    </column>
    <item row="0" column="1">
     <property name="text">
      <string/>
     </property>
     <property name="font">
      <font>
       <pointsize>10</pointsize>
      </font>
     </property>
    </item>
   </widget>
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>470</x>
      <y>250</y>
      <width>221</width>
      <height>501</height>
     </rect>
    </property>
    <property name="title">
     <string>Manual MEMS control</string>
    </property>
    <widget class="QGroupBox" name="groupBox_2">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>201</width>
       <height>151</height>
      </rect>
     </property>
     <property name="title">
      <string>Move MEMS</string>
     </property>
     <widget class="QLabel" name="label_6">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>40</y>
        <width>51</width>
        <height>22</height>
       </rect>
      </property>
      <property name="text">
       <string>Piston</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="label_7">
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>40</y>
        <width>31</width>
        <height>22</height>
       </rect>
      </property>
      <property name="text">
       <string>Tip</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="label_8">
      <property name="geometry">
       <rect>
        <x>140</x>
        <y>40</y>
        <width>31</width>
        <height>22</height>
       </rect>
      </property>
      <property name="text">
       <string>Tilt</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QPushButton" name="piston_up">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>70</y>
        <width>31</width>
        <height>30</height>
       </rect>
      </property>
      <property name="text">
       <string>Up</string>
      </property>
     </widget>
     <widget class="QPushButton" name="piston_down">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>110</y>
        <width>51</width>
        <height>30</height>
       </rect>
      </property>
      <property name="text">
       <string>Down</string>
      </property>
     </widget>
     <widget class="QPushButton" name="tip_up">
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>70</y>
        <width>31</width>
        <height>30</height>
       </rect>
      </property>
      <property name="text">
       <string>Up</string>
      </property>
     </widget>
     <widget class="QPushButton" name="tip_down">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>110</y>
        <width>51</width>
        <height>30</height>
       </rect>
      </property>
      <property name="text">
       <string>Down</string>
      </property>
     </widget>
     <widget class="QPushButton" name="tilt_up">
      <property name="geometry">
       <rect>
        <x>140</x>
        <y>70</y>
        <width>31</width>
        <height>30</height>
       </rect>
      </property>
      <property name="text">
       <string>Up</string>
      </property>
     </widget>
     <widget class="QPushButton" name="tilt_down">
      <property name="geometry">
       <rect>
        <x>130</x>
        <y>110</y>
        <width>51</width>
        <height>30</height>
       </rect>
      </property>
      <property name="text">
       <string>Down</string>
      </property>
     </widget>
    </widget>
    <widget class="QGroupBox" name="groupBox_3">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>190</y>
       <width>201</width>
       <height>251</height>
      </rect>
     </property>
     <property name="title">
      <string>Presets</string>
     </property>
     <widget class="QLabel" name="label_12">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>30</y>
        <width>31</width>
        <height>22</height>
       </rect>
      </property>
      <property name="text">
       <string>Off</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="label_13">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>70</y>
        <width>31</width>
        <height>22</height>
       </rect>
      </property>
      <property name="text">
       <string>On</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="label_14">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>110</y>
        <width>31</width>
        <height>22</height>
       </rect>
      </property>
      <property name="text">
       <string>Flat</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QPushButton" name="off_set">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>30</y>
        <width>51</width>
        <height>30</height>
       </rect>
      </property>
      <property name="text">
       <string>Set</string>
      </property>
     </widget>
     <widget class="QPushButton" name="off_restore">
      <property name="geometry">
       <rect>
        <x>130</x>
        <y>30</y>
        <width>61</width>
        <height>30</height>
       </rect>
      </property>
      <property name="text">
       <string>Restore</string>
      </property>
     </widget>
     <widget class="QPushButton" name="on_set">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>70</y>
        <width>51</width>
        <height>30</height>
       </rect>
      </property>
      <property name="text">
       <string>Set</string>
      </property>
     </widget>
     <widget class="QPushButton" name="on_restore">
      <property name="geometry">
       <rect>
        <x>130</x>
        <y>70</y>
        <width>61</width>
        <height>30</height>
       </rect>
      </property>
      <property name="text">
       <string>Restore</string>
      </property>
     </widget>
     <widget class="QPushButton" name="flat_set">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>110</y>
        <width>51</width>
        <height>30</height>
       </rect>
      </property>
      <property name="text">
       <string>Set</string>
      </property>
     </widget>
     <widget class="QPushButton" name="flat_restore">
      <property name="geometry">
       <rect>
        <x>130</x>
        <y>110</y>
        <width>61</width>
        <height>30</height>
       </rect>
      </property>
      <property name="text">
       <string>Restore</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_15">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>150</y>
        <width>67</width>
        <height>22</height>
       </rect>
      </property>
      <property name="text">
       <string>Preset</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLineEdit" name="preset_path">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>170</y>
        <width>171</width>
        <height>30</height>
       </rect>
      </property>
     </widget>
     <widget class="QPushButton" name="preset_restore">
      <property name="geometry">
       <rect>
        <x>110</x>
        <y>210</y>
        <width>61</width>
        <height>30</height>
       </rect>
      </property>
      <property name="text">
       <string>Restore</string>
      </property>
     </widget>
     <widget class="QPushButton" name="preset_save">
      <property name="geometry">
       <rect>
        <x>50</x>
        <y>210</y>
        <width>51</width>
        <height>30</height>
       </rect>
      </property>
      <property name="text">
       <string>Save</string>
      </property>
     </widget>
    </widget>
    <widget class="QLabel" name="label_16">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>460</y>
       <width>141</width>
       <height>22</height>
      </rect>
     </property>
     <property name="text">
      <string>Segment (0 for all)</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="segment_selection">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>460</y>
       <width>41</width>
       <height>30</height>
      </rect>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_4">
    <property name="geometry">
     <rect>
      <x>700</x>
      <y>250</y>
      <width>241</width>
      <height>561</height>
     </rect>
    </property>
    <property name="title">
     <string>Nuller optimisation</string>
    </property>
    <widget class="QGroupBox" name="groupBox_5">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>191</width>
       <height>211</height>
      </rect>
     </property>
     <property name="title">
      <string>TT optimisation</string>
     </property>
     <widget class="QPushButton" name="tt_opt">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>30</y>
        <width>171</width>
        <height>30</height>
       </rect>
      </property>
      <property name="text">
       <string>Do TT optimisation</string>
      </property>
     </widget>
     <widget class="QGraphicsView" name="Injectionmap">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>70</y>
        <width>171</width>
        <height>131</height>
       </rect>
      </property>
     </widget>
    </widget>
    <widget class="QGroupBox" name="groupBox_6">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>250</y>
       <width>221</width>
       <height>301</height>
      </rect>
     </property>
     <property name="title">
      <string>Nulling Optimisation</string>
     </property>
     <widget class="QPushButton" name="null_opti">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>260</y>
        <width>171</width>
        <height>30</height>
       </rect>
      </property>
      <property name="text">
       <string>Do Nuller optimisation</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="seg_move">
      <property name="geometry">
       <rect>
        <x>160</x>
        <y>40</y>
        <width>51</width>
        <height>30</height>
       </rect>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
     <widget class="QLabel" name="label">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>40</y>
        <width>131</width>
        <height>22</height>
       </rect>
      </property>
      <property name="text">
       <string>Segment to move</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_2">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>80</y>
        <width>121</width>
        <height>22</height>
       </rect>
      </property>
      <property name="text">
       <string>Null to scan</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="null_scan">
      <property name="geometry">
       <rect>
        <x>160</x>
        <y>80</y>
        <width>51</width>
        <height>30</height>
       </rect>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
     <widget class="QLabel" name="label_3">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>120</y>
        <width>121</width>
        <height>22</height>
       </rect>
      </property>
      <property name="text">
       <string>Scan wait</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="scan_wait">
      <property name="geometry">
       <rect>
        <x>160</x>
        <y>120</y>
        <width>51</width>
        <height>30</height>
       </rect>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
     <widget class="QLineEdit" name="num_loops">
      <property name="geometry">
       <rect>
        <x>160</x>
        <y>160</y>
        <width>51</width>
        <height>30</height>
       </rect>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
     <widget class="QLabel" name="label_4">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>160</y>
        <width>131</width>
        <height>22</height>
       </rect>
      </property>
      <property name="text">
       <string>Number of loops</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_5">
      <property name="geometry">
       <rect>
        <x>40</x>
        <y>200</y>
        <width>131</width>
        <height>22</height>
       </rect>
      </property>
      <property name="text">
       <string>Null scan range</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="null_scan_range">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>220</y>
        <width>181</width>
        <height>30</height>
       </rect>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1130</width>
     <height>27</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
